core/vm/evm.go

```diff
@@ -432,17 +445,19 @@ func (evm *EVM) create(caller ContractRef, code []byte, gas uint64, value *big.I
 
 // Create creates a new contract using code as deployment code.
 func (evm *EVM) Create(caller ContractRef, code []byte, gas uint64, value *big.Int) (ret []byte, contractAddr common.Address, leftOverGas uint64, err error) {
+	codeAndHash := codeAndHash{code: code}
 	contractAddr = crypto.CreateAddress(caller.Address(), evm.StateDB.GetNonce(caller.Address()))
-	return evm.create(caller, code, gas, value, contractAddr)
+	return evm.create(caller, codeAndHash, gas, value, contractAddr)
```

Comment:
You can probably inline `codeAndHash{code: code}` here., a bit shorter/cleaner (imho).